### PR TITLE
Tag component tweaks to get rid of hardcoded tag variables

### PR DESCRIPTION
**Pull request by teodosii**
# What this PR does

- Removed the usage of `var(--` within the Tag component to help get rid of the vars file once we fully migrate to emotion
- Added few other display tweaks and migrated a few stylesheets to emotion

### CODE REVIEW

File: grafana-plugin/src/styles/utils.styles.ts [hunk 1]
```diff
@@ -24,6 +29,34 @@ export const getUtilStyles = (_theme: GrafanaTheme2) => {
   };
 };
 
+export function getLabelBackgroundTextColorObject(
```
Comment (teodosii) [inline grafana-plugin/src/styles/utils.styles.ts:32]: This part is actually the core of `Badge` component in grafana but changing `Tag` to be replaced to `Badge` in all our usecases would have taken a lot more time, and instead I preferred this approach where we rely on tinycolor to generate these (same as grafana)

Comment (maskin25) [reply grafana-plugin/src/styles/utils.styles.ts:32]: is there any helper in `@grafana/ui` package that returns such object? I think it could be there...

Comment (maskin25) [reply grafana-plugin/src/styles/utils.styles.ts:32]: I found several helpers: `getTagColorsFromName, getTextColorForBackground, getTextColorForAlphaBackground` etc, maybe the can be used here

File: grafana-plugin/src/styles/utils.styles.ts [hunk 2]
```diff
@@ -39,5 +72,6 @@ export const bem = (...args: string[]) =>
 export enum COLORS {
   ALWAYS_GREY = '#ccccdc',
   GRAY_8 = '#595959',
+  GRAY_9 = '#434343',
```
Comment (maskin25) [inline grafana-plugin/src/styles/utils.styles.ts:75]: can we get such color values from the `theme`?

File: grafana-plugin/src/styles/utils.styles.ts (hunk 1 above)
Comment (teodosii) [reply grafana-plugin/src/styles/utils.styles.ts:32]: Neither of them is closer to what this does :(

**Review by brojd — APPROVED** [review]

**Review by maskin25 — APPROVED** [review]

**Pull request merged**